### ISSUE TITLE
Bon, et maintenant on fait quoi ? ^^

### ISSUE THREAD

**Issue by Becheler** [stdupas/EnvironmentalDemogeneticsABC#10]
open !

**Comment by Becheler** [comment]
L'idée c'est que maintenant la branche develop est fonctionnelle. C'est à elle qu'on va rajouter des fonctionnalités. On se place sur dévelop, on ouvre une nouvelle branche (git checkout -b feature) du nom de la feature qu'on veut rajouter, on fait nos premiers commits dessus, et après test, débugage et quelques rebases (pour suivre quand même l'évolution de dévelop) on fait un true merge final avec l'option --no-ff si besoin est (autrement dit si develop n'a pas évolué entre temps).

Pour master, je vais essayer de voir si je peux la merger de manière pas trop crado avec develop... aucune idée de ce que ça va donner. 

Pour nicheFunctions je suis d'avis de la laisser tomber. Le graph commence à être super dégueulasse, j'ai beaucoup de mal à retrouver les commits qui sont ou pas dans develop. Mais quand je compare les fichiers entre branches, les différences ont l'air d'être superflues.
Voili voilou :)

**Comment by Becheler** [comment]
Surtout ne jamais commiter directement sur develop. Chaque fois on ouvre une autre branche, locale ou distante selon qu'on taff dessus resp tout seul ou à deux :)

**Comment by stdupas** [comment]
ok je me mets sur develop maintenant et je peux créer mes petites fonction forward dans un fichier Forward.R

**Comment by stdupas** [comment]
comment je fais pour me mettre sur develop au fait?

**Comment by stdupas** [comment]
git branch develop
...?

**Comment by stdupas** [comment]
après je pull

**Comment by Becheler** [comment]
Il faut d'abord que tu la pull, et que tu la track. Là c'est le moment où j'ouvre discrètement un onglet google pour me rappeler comment on fait pour pas faire de bêtise... ^^"

`git checkout -b develop origin/develop`

Du coup le checkout -b va à la fois créer ta branche locale et t'y switcher. Du coup quand tu es sur develop tu fais un :

`git checkout -b forward`

Et tu commit sur cette branche forward locale.

**Comment by stdupas** [comment]
ça c'est cool, j'ai failli faire une bettise. "Le reflexe chercher par soi même !" je l 'ai pas

**Comment by stdupas** [comment]
ça marchait pas : git checkout -b develop origin/develop
fatal: git checkout: updating paths is incompatible with switching branches.
Did you intend to checkout 'origin/develop' which can not be resolved as commit?

J'ai du faire 
git fetch
pour updater mes branches avant de chackout
git checkout develop

**Comment by Becheler** [comment]
Alors en ce qui concerne pour la progression de développement sur la branche `forward` :
tu as fait quelques commit (concernant uniquement le forward) pour ta fonctionnalité forward et tu veux suivre l'évolution de la branche develop pour être certain que ton code continue à fonctionner : en étant sur ta branche forward tu fais de temps à autre un

`git rebase develop` 

Et quand ta série de modification sur la branche forward est terminée, tu fais un ultime rebase pour avoir les tous derniers update de develop, tu test le code pour traquer les bugs, et tu partages ta fonctionnalité forward en la mergeant. Pour ce faire :

`git checkout develop` pour retourner sur ta branche develop
`git merge --no-ff forward` pour avoir un historique impec

Et après il est préférable d'effacer ta vieille branche forward qui ne sert plus à rien. Tu te place sur develop et :

`git branch -d forward`

et si ta branche forward avait été aussi envoyée aux copains :

`git push origin :forward` qui veut en fait dire que tu envoies une branche vide sur forward (rien avant les deux points...)

Et tadzam sauf bêtise de ma part :)

**Comment by Becheler** [comment]
Oki cool si t'as pu le faire ! :+1: 
Par curiosité, t'as fait un `git remote update` avant le `git fetch` ?

**Comment by stdupas** [comment]
non? j'ai pas remote updaté

**Comment by Becheler** [comment]
Oki. ça a marché ? C'était là en fait :
http://stackoverflow.com/questions/945654/git-checkout-on-a-remote-branch-does-not-work

**Comment by Becheler** [comment]
Encore une énigme de Git-le-monstrueux... ^^

**Comment by stdupas** [comment]
oui j'ai des fichiers datés de 18:20 ça doit être les bons. remote update permet de fetcher sur tous les remote branch, la y en avait qu'une , c'est ça?

**Comment by Becheler** [comment]
No idea Sir !

**Comment by Becheler** [comment]
Apparemment t'as bon : http://stackoverflow.com/questions/1856499/differences-between-git-remote-update-and-fetch

**Comment by stdupas** [comment]
dans ton main sur develop, je peux remplacer GeneticData <- read.table("GeneticData.txt", header=TRUE)
par 
load("GeneticData.RData")
(j'ai pas ce fichier GeneticData.txt)

**Comment by Becheler** [comment]
Yep. Bah disons qu'à la base j'avais fait une bếtise en l'enregistrant en fichier binaire. Vaut mieux l'avoir au format texte, en plus il pourra également être placé sous suivi de version. Enfin je pense :)

**Comment by Becheler** [comment]
En plus l'ancien format est obsolète, vu qu'il incorporait les numéro de cellules, qu'on peut en fait retrouver grâce à cellFromXY...
